google/cloud/aiplatform/training_job_base.py

```diff
@@ -0,0 +1,262 @@
+from typing import Optional, Sequence
+
+import logging
+import time
+import sys
+
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import models
+from google.cloud.aiplatform import datasets
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform import utils
+
+from google.cloud.aiplatform_v1beta1.services.pipeline_service import (
+    client as pipeline_service_client,
+)
+
+from google.cloud.aiplatform_v1beta1.types import io as gca_io
+from google.cloud.aiplatform_v1beta1.types import model as gca_model
+from google.cloud.aiplatform_v1beta1.types import (
+    training_pipeline as gca_training_pipeline,
+)
+from google.cloud.aiplatform_v1beta1.types import pipeline_state as gca_pipeline_state
+
+from google.rpc import code_pb2
+
+logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+_LOGGER = logging.getLogger(__name__)
+
+_PIPELINE_COMPLETE_STATES = set(
+    [
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_SUCCEEDED,
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_FAILED,
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_CANCELLED,
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_PAUSED,
+    ]
+)
+
+class TrainingJob(base.AiPlatformResourceNoun):
+    client_class = pipeline_service_client.PipelineServiceClient
+    _is_client_prediction_client = False
+
+    def __init__(
+        self,
+        display_name: str,
+        container_uri: str,
+        requirements: Optional[Sequence[str]] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        super().__init__(project=project, location=location, credentials=credentials)
+        self._display_name = display_name
+        self._container_uri = container_uri
+        self._requirements = requirements
+        self._project = project
+        self._credentials = credentials
+        self._gca_resource = None
+
+    def _create_input_data_config(
+        self,
+        dataset: Optional[datasets.Dataset],
+        training_fraction_split: float,
+        validation_fraction_split: float,
+        test_fraction_split: float,
+        base_output_dir: Optional[str] = None,
+    ) -> gca_training_pipeline.InputDataConfig:
+        input_data_config = None
+
+        if dataset:
+            # Create fraction split spec
+            fraction_split = gca_training_pipeline.FractionSplit(
+                training_fraction=training_fraction_split,
+                validation_fraction=validation_fraction_split,
+                test_fraction=test_fraction_split,
+            )
+
+            # create input data config
+            input_data_config = gca_training_pipeline.InputDataConfig(
+                fraction_split=fraction_split,
+                dataset_id=dataset.name,
```

Comment:
@sasha-gitg The only thing `dataset` is used for is to get the name, should the name just be passed in directly? I'll leave it as a dataset object for now.

Reply:
Eventually we should accept either a Dataset object, Dataset resource name, or a Dataset resource id though that doesn't have to be added in this PR.

Reply:
Got it.